src/slack_model.jl

```diff
@@ -68,16 +68,23 @@ function slack_meta(meta :: NLPModelMeta)
   )
 end
 
-"Construct a `SlackModel` from another type of model."
-function SlackModel(model :: AbstractNLPModel)
+"""
+    ConvertToSlackModel(model)
+
+Convert an NLPModel or NLSModel to a SlackModel, without using specialized information
+about the type of `model`.
+
+You may want `SlackModel(model)` instead.
+"""
+function ConvertToSlackModel(model :: AbstractNLPModel)
```

Comment:
Why would we ever want this?

Reply:
I'm not sure we will but currently, it's used for testing on `test/test_slack_model.jl`. Possibly won't need it after this is finished.